reactioncommerce/reaction
# feat: authorization plugin & checkPermissions functionality

Impact: **minor**  
Type: **feature|refactor**

## Issue
Our authorization / permission related code is currently living inside of our `accounts` package. With the work being done to incorporate `ory keto` as an authorization service, we need to break out these functions into their own package.

- for #5736
- for #5737

## Solution
1. Create a core-services `simple-authorization` package, and move authorization functionality from `accounts` into this new package
1. Create a `checkPermissions` function to replace `userHasPermission` in most places. Since we were essentially duplicating all `ReactionError` code throughout the app, this update throws the error inside `checkPermissions` so there is no need to throw it everywhere.

## Testing
1. Load the app and see that it works as it should. This change should change nothing, aside from the locations of some files. Specifically, you can:
- Try to admin an order or a product from a user that isn't allowed, see it doesn't work
- Try to admin an order or product from a user that is allowed, see it works
